### PR TITLE
Added exceptions for null; added catch for unhandled promise rejection

### PR DESCRIPTION
**Pull request by bryce-seifert**
Add replacements for Speed, ClipID, and SlotID button variables in order to be more descriptive than 0 or null

### CODE REVIEW

**Comment by sphlabs** [comment]
Given that clip and slot IDs are only 1 or 2 (maybe 3) characters, I think a long text string would often end up displaying badly, especially if one has set a fixed font size.

Perhaps a single 'null' character would be more appropriate here? I don't know if there is a meaningful ascii/unicode character that represents null/nil, but even an 'X' could suffice.

Also, with speed, I'm not sure if 'stopped' is more meaningful than '0'? I suppose it may depend how the variable is used, but in the case of something like `Speed:\n$(hypderdeck:speed)` I think 0 would make more sense.

**Comment by bryce-seifert** [comment]
I agree with you on speed, 0 likely makes more sense.

I think the issue I was trying to solve is the times when multiple button variables are returning 0 or null at the same time, it can be confusing which button is which.

I changed the longer strings to "Empty" for slotId and "None" for clipId since those are closer in length to null, yet I think still distinguishable from each other what they are representing.

**Comment by haakonnessjoen** [comment]
I am not sure if i like this. I like variables that are numeric to stay numeric. Maybe most because we want to be able to use variables for much more than just buttons in the future.

If you put "Clip:\n${hd:clipId}" as the label on your button, you will never be unsure of what the numbers mean.

**Comment by sphlabs** [comment]
I'm inclined to agree, although I think it could be better than at present, which is 'null'.

Options:
'' (blank) - it's simple and conveys meaning by its absence
'X' (or some other single character) - has positive confirmation of a null value

I don't think it should be '0' as zero has a different meaning to null.

**Comment by bryce-seifert** [comment]
Blank seems like a good, simple option. If there's a way to affect that button text without inherently changing the variable value from numeric, that'd be great.

I understand the desire to keep variables numeric, but from a UX / legibility standpoint having nulls everywhere on buttons is not the most desirable.

**Comment by haakonnessjoen** [comment]
Sure, '' or 0 for absence of number is fine. From a UX standpoint, we could improve variables by adding a conditional default value when used.

**Comment by sphlabs** [comment]
Having thought about this some more, I think the correct solution is to display something that actively represents 'null' as positive feedback, which '' doesn't do, and makes it distinct to no feedback at all (e.g. when not connected).

My suggestion is an em dash, —, Unicode: U+2014, UTF-8: E2 80 94

**Comment by bryce-seifert** [comment]
That feels right to me too, good idea. Updated to request to reflect that.

File: index.js [hunk 1]
```diff
@@ -1001,24 +1001,40 @@ class instance extends instance_skel {
...
 		variables.push({
 			label: 'Play speed',
 			name:  'speed'
 		});
 		this.setVariable('speed', this.transportInfo['speed']);
 
+		//Clip ID and Slot ID  null exceptions
+		
+		let clipIdVariable = '—';
+		if (this.transportInfo['clipId'] == null) {
```
Comment (krocheck) [inline index.js]: Do you want to do a `... != null` and skip the else for both of these?

Comment (bryce-seifert) [reply index.js]: Got it, done!

File: index.js [hunk 2]
```diff
@@ -1001,24 +1001,35 @@ class instance extends instance_skel {
...
 
+		//Clip ID and Slot ID  null exceptions
+		
+		let clipIdVariable = '—';
+		if (this.transportInfo['clipId'] != null) {
+			clipIdVariable = this.transportInfo['clipId'];
+		}
+		let slotIdVariable = '—';
+		if (this.transportInfo['slotId'] != null) {
+			slotIdVariable = '—';
```
Comment (krocheck) [inline index.js]: This is now missing its proper assign statement: `slotIdVariable = this.transportInfo['slotId'];`

Comment (bryce-seifert) [reply index.js]: Ah, my bad. I think I fixed it.

**Pull request merged**